service/history/task/cross_cluster_task.go

```diff
@@ -73,9 +76,35 @@ type (
 		taskProcessor   Processor       //TODO: wire up the dependency
 		redispatchFn    func(task Task) //TODO: wire up the dependency
 		maxRetryCount   dynamicconfig.IntPropertyFn
+		settable        future.Settable
 	}
 )
 
+// NewCrossClusterTaskForTargetCluster creates a CrossClusterTask
+// for the processing logic at target cluster
+// the returned the Future will be unblocked when after the task
+// is processed. The future value has type types.CrossClusterTaskResponse
+// and there will be not error returned for this future. All errors will
+// be recorded by the FailedCause field in the response.
+func NewCrossClusterTaskForTargetCluster(
+	shard shard.Context,
+	taskRequest *types.CrossClusterTaskRequest,
+	logger log.Logger,
+	maxRetryCount dynamicconfig.IntPropertyFn,
+) (CrossClusterTask, future.Future) {
```

Comment:
What do you think to have the future return in all NewCrossClusterXXXTask() and no need for the NewCrossClusterTaskForTargetCluster?

Reply:
Yeah, I agree that future can be returned in all NewCrossClusterXXXTask() functions. But I guess there should still be a dedicated constructor for tasks in target cluster as the input parameter will be quite different? Either the current `CrossClusterTaskRequest` or some more detailed fields like signal input, childworkflow start attribute, task state etc. For tasks in the source cluster, those won't be necessary as they can be loaded from mutable state or events cache in source cluster. But in target cluster, we have to pass them in when creating the task.